### ISSUE TITLE
Cookies are not deleted in Firefox Quantum when Multi-Account Containers add-on is installed

### ISSUE THREAD

**Issue by babageorge** [mrdokenny/Cookie-AutoDelete#291]
### Describe the Bug/Feature
I have used Cookie AutoDelete in Firefox 57 with automatic cleaning enabled, a delay set to 0.01 min, and a notification show after cookie clean-up. And it has worked very well. Recently, however, I installed the Multi-Account Containers add-on and tried to browse the net with the installed containers. I noticed that in this case after closing the tabs/containers, the automatic cleaning was not performed. This happens when there are either multiple tabs of the same container or tabs of various containers open. On the contrary, when I browse the net in the default container, then Cookie AutoDelete has no problem.

### Steps for anyone to reproduce the bug
1. Download and install Cookie Auto Delete add-on in Firefox 57.
2. Enable auto-clean. Set delay other than 0 min. Set notification after cookie cleanup.
3. Check performance of add-on.
4. Download and install Multi-Account Containers add-on.
5. Open various tabs in different containers. Upon closing the tabs or the containers check whether Cookie Auto Delete erases the cookies in Preferences/Privacy & Security/Show Cookies.

### Your settings 

- OS/version: Linux Mint 18.3 cinnamon edition (kernel 4.10.0-42-generic)
- Browser/version: Firefox Quantum/57.0.1 (64-bit) (mint-1.0)
- Cookie AutoDelete version: 2.0.1
- Other installed extensions: AdNauseam 3.4.402; Bulk Media Downloader 0.2.1; Decentraleyes 2.0.1; HTTPS Everywhere 2017.12.6; Multi-Account Containers 4.1.0; NoScript 10.1.5.9; Print Edit WE 20.1; YouTube MP4 Downloader 2.0.7

**Comment by kennethtran93** [comment]
Perhaps a duplicate issue of #197.

**Comment by babageorge** [comment]
The workaround which iis given by means of creating empty expressions in all the installed containers. I can confirm that

**Comment by babageorge** [comment]
Sorry for not checking previous entries of similar issue. The workaround that is given in issue #197, by means of adding an empty whitelist expression in each container and then deleting it, worked for me too! I don't know, however, if this issue/bug(?) should be closed so easily.

**Comment by babageorge** [comment]
Sorry for not checking previous entries for similar issues. The workaround that is given in entry #197, by means of adding an empty whitelist expression in each container and then deleting it, worked for me too! I don't know, however, if this issue/bug(?) should be closed so easily.

**Comment by saberworks** [comment]
I was pulling my hair out trying to figure out why most cookies were not being deleted.  The workaround specified above (which I did not see in #197 for some reason?) worked.  I used the settings pane of the Cookie AutoDelete to ensure every container had at least one whitelist entry.  Now all the "clean" actions work.  Note that it wasn't just "autoclean" that was not working.  Manually pressing the "clean" button would not work either.  It just would not delete any cookies, even if they were not whitelisted or graylisted.

I also am not convinced that Firefox itself is behaving properly with regards to cookies.  I have specific sites that are set to only allow cookies for session (and all permutations of the domain and http vs https) but firefox continues to allow cookies that that expire a year in the future for this domain.  And the "session" cookies never get deleted if the browser is set to restore tabs on startup (even if there are no tabs open for this specific site).  Thanks for creating this addon.
